packages/functional-tests/pages/settings/components/unitRow.ts

```diff
@@ -135,6 +135,12 @@ export class DataCollectionRow extends UnitRow {
       .getAttribute('aria-checked');
   }
 
+  async isToggleSwitch() {
+    const toggle = this.page.locator('[data-testid=metrics-opt-out]');
+    await toggle.waitFor({ state: 'visible' });
+    return toggle.isVisible();
```

Comment:
Does the `waitFor` not already check if the locator is visible?

nit. looks like state = visible is the default, so no need to specify it.